subprojects/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/inputs/undeclared/UndeclaredBuildInputsIgnoringIntegrationTest.groovy

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.configurationcache.inputs.undeclared
+
+import org.gradle.configurationcache.AbstractConfigurationCacheIntegrationTest
+import org.gradle.initialization.StartParameterBuildOptions
+import org.gradle.test.fixtures.file.TestFile
+
+class UndeclaredBuildInputsIgnoringIntegrationTest extends AbstractConfigurationCacheIntegrationTest {
+    def 'can ignore a file system check configuration inputs #location'() {
+        given:
+        def inputFile = fileClosure.call(testDirectory)
+        buildFile("""
+            println("exists = " + new File("${inputFile.absolutePath}").exists())
+        """)
+
+        when:
+        configurationCacheRun()
+
+        then:
+        outputContains("exists = false")
+        problems.assertResultHasProblems(result) {
+            withInput("Build file 'build.gradle': file system entry")
+        }
+
+        when:
+        file("gradle.properties") << """$IGNORE_FS_CHECKS_PROPERTY=$pattern"""
+        configurationCacheRun()
+
+        then:
+        outputContains("exists = false")
+        problems.assertResultHasProblems(result) {
+            withNoInputs()
+        }
+
+        where:
+        location                        | pattern            | fileClosure
+        "inside the project directory"  | "build/*.lock"     | { TestFile testDir -> testDir.file("build/test.lock").absoluteFile }
+        "outside the project directory" | "../../*.lock"     | { TestFile testDir -> testDir.file("../../test.lock").absoluteFile }
```

Comment:
Well, yes, checking for the existence of these files can lead to flakiness, indeed. 
But I think passing all these kinds of paths in the integration tests has some value of its own, as it also tests that the paths are not corrupted somewhere between the property and the implementation.
I think we could override the test directory so that its parents are under our control.

Reply:
I tried to set up a project directory in a way that it is safe to look at its parent directories, but the test infrastructure did not let me do that properly (the working director of the runner is still the same, and the paths are actually relativized against the working directory). I decided to remove the test cases that interact with the file system outside the test directory for now.